app/Metrics/WikiEntityImports.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+
+namespace App\Metrics;
+
+use LKDevelopment\HorizonPrometheusExporter\Contracts\Exporter;
+use Prometheus\CollectorRegistry;
+use App\WikiEntityImport;
+use App\WikiEntityImportStatus;
+
+class WikiEntityImports implements Exporter
+{
+    protected $pending;
+    protected $successful;
+    protected $failed;
+
+    public function metrics(CollectorRegistry $collectorRegistry)
+    {
+        $this->pending = $collectorRegistry->getOrRegisterGauge(
+            config('horizon-exporter.namespace'),
+            'wiki_entity_imports_pending',
+            'The number of pending Entity imports currently being processed.',
+        );
+        $this->successful = $collectorRegistry->getOrRegisterGauge(
```

Comment:
It's a bit unfortunate that both this as well as `failed` are gauges when instead they should be counters. However, the library in use does not expose an option for calling SET on a counter, instead it only knows about INCREASE, which would require knowledge about the previous state in this scenario.

Reply:
Could we inject a collector registry into the controller and call `increase` there?

https://github.com/wbstack/api/pull/830/files#diff-29372c16f0d76454ce93dfa22a7e0833e4ae2a7a404e3adee053364af885a060R79

I'm not sure if this is excessively abusing the "horizon" nature of the exporter but this would be more "natural" than a gauge perhaps. Although, I think given that currently we don't have a plan to garbage collect old Statuses it's not that bad to treat this as a gauge

Reply:
Should be worth a try.

Reply:
This is now implemented.